src/java/org/apache/cassandra/config/Config.java

```diff
@@ -905,6 +905,8 @@ public static void setClientMode(boolean clientMode)
     public volatile boolean non_partition_restricted_index_query_enabled = true;
     public volatile int sai_sstable_indexes_per_query_warn_threshold = 32;
     public volatile int sai_sstable_indexes_per_query_fail_threshold = -1;
+    public volatile boolean intersect_filtering_query_warned = true;
+    public volatile boolean intersect_filtering_query_enabled = true;
```

Comment:
When we actually fix CASSANDRA-19007, this guardrail should just be removed, as it will no longer be necessary. In light of this, I haven't added an entry in `cassandra.yaml`.